### PR TITLE
Add RBAC migration documentation

### LINKED ISSUE #3619: Update Wazuh documentation with the new changes introduced with the RBAC migration
@CarlosRS9 commented on [Fri Dec 18 2020](https://github.com/wazuh/wazuh/issues/6966)

Hello team,

We need to add or update our Wazuh documentation for anything related to the changes in the orm and also add a new page explaining how the rbac database migration works.

### PR DESCRIPTION
**Pull request by CarlosRS9**
Hello team.

This PR closes #3619. Here is a list with the changes applied:

- Updated RBAC `How it works documentation` page with a new section called RBAC resource types
- Renamed `Actions, resources and effects` section in the `How it works` to `RBAC Policies` to avoid confusing users.
- Added a new page named `RBAC database integrity` to explain how the integrity and migration works.
- Minor fixes to the `How it works` page.
- Minor fixes to the `security_resources` CLI page.

### CODE REVIEW

File: source/user-manual/api/rbac/how-it-works.rst [hunk 1]
```diff
@@ -40,3 +40,17 @@ RBAC in Wazuh can be configured in two different and opposite ways: **black** an
...
 - **Black list mode:** everything is allowed. The administrator configures roles to restrict permissions.
+
+RBAC resource types
+-------------------
+
+Every RBAC resource (policies, roles, rules and users) can be classified regarding its resource or protection type used to determine whether the Wazuh API should be able to remove or modify that given RBAC resource. There are 3 different resource types:
+
+- **User resources:** A regular RBAC resource created using the Wazuh API. It is NOT protected and can be modified or removed using the Wazuh API by any user as long as it belongs to the appropriate RBAC Role. Check :ref:`RBAC reference <api_rbac_reference>` to learn more about the actions required to create RBAC resources.
+
+- **Protected resources:** A regular RBAC resource. It is protected so it cannot be modified or removed using the Wazuh API even if the user has the appropriate RBAC role. It must be manually created, updated or removed by using the :ref:`security_resources <security_resources>` tool instead.
```
Comment (davidjiglesias) [inline source/user-manual/api/rbac/how-it-works.rst:51]:
```suggestion
- **Protected resources:** A special RBAC resource. It is protected so it cannot be modified or removed using the Wazuh API even if the user has the appropriate RBAC role. It must be manually created, updated or removed by using the :ref:`security_resources <security_resources>` tool instead.
```

File: source/user-manual/api/rbac/how-it-works.rst [hunk 2]
```diff
@@ -40,3 +40,17 @@ RBAC in Wazuh can be configured in two different and opposite ways: **black** an
...
+
+- **Default resources:** A default Wazuh RBAC resource. It cannot be modified or removed by any user. The default resources are managed by Wazuh and could be added, updated or removed in different Wazuh releases. Neither the Wazuh API nor the CLI tool are able to update or remove the default resources. However, they can be used by the user as any other resource. As an example, it is possible to create a custom Role and assign to it a default policy.
+
+.. note::
+    Any RBAC policy, role, rule or user created using the Wazuh API will be created as `user` type. The :ref:`security_resources <security_resources>` tool can be used to create, modify or remove `protected` resources.
```
Comment (davidjiglesias) [inline source/user-manual/api/rbac/how-it-works.rst:56]:
```suggestion
    Any RBAC policy, role, rule or user created using the Wazuh API will be created with the `user` type. The :ref:`security_resources <security_resources>` tool can be used to create, modify or remove `protected` resources.
```

File: source/user-manual/api/rbac/rbac-database-integrity.rst [hunk 3]
```diff
@@ -0,0 +1,28 @@
+.. Copyright (C) 2021 Wazuh, Inc.
```
Comment (davidjiglesias) [inline source/user-manual/api/rbac/rbac-database-integrity.rst:1]: Move this new file to development documentation, not here, I believe this is not user related documentation.

**Review by davidjiglesias — CHANGES_REQUESTED** [review]

File: source/user-manual/api/rbac/rbac-database-integrity.rst (hunk 3 above)
Comment (CarlosRS9) [reply source/user-manual/api/rbac/rbac-database-integrity.rst]: done

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**